packages/iov-bns/src/bnsconnection.spec.ts

```diff
@@ -321,6 +322,43 @@ describe("BnsConnection", () => {
     connection.disconnect();
   });
 
+  it("can register a username", async () => {
+    pendingWithoutBnsd();
+    const connection = await BnsConnection.establish(bnsdTendermintUrl);
+    const chainId = await connection.chainId();
+
+    const profile = new UserProfile();
+    const wallet = profile.addWallet(Ed25519HdWallet.fromEntropy(await Random.getBytes(32)));
+    const identity = await profile.createIdentity(wallet.id, HdPaths.simpleAddress(0));
+
+    // Create and send registration
+    const address = keyToAddress(identity.pubkey);
+    const username = `testuser`; // _${Math.random()}`;
+    const registration: RegisterUsernameTx = {
+      kind: TransactionKind.RegisterUsername,
+      chainId: chainId,
+      signer: identity.pubkey,
+      addresses: new Map([[chainId, address]]),
```

Comment:
This is a good question for me, what the address should be here, why bytes is human readable encoding, etc...

Reply:
Good question indeed. What happens when one chain supports different address types? I tend to think we should store the user's choice.

For some broken chains out there, there is no proper binary address.